src/proofPointValidator.ts

```diff
@@ -0,0 +1,124 @@
+import { EthereumAddress, ProofPointId } from "./ethereumProofPointEvent";
+import { ProofPointValidateResult } from "./proofPointValidateResult";
+import {
+  GeneralProofPointResolver,
+  ProofPointResolver,
+} from "./proofPointResolver";
+import { ProofPointStatus } from "./proofPointStatus";
+import {
+  GeneralProofPointAuthenticator,
+  ProofPointAuthenticator,
+} from "./proofPointAuthenticator";
+import { ethers } from "ethers";
+import { EthereumProofPointRegistryRoot } from ".";
+import { RealHttpClient } from "./httpClient";
+import { EthereumAddressResolver } from "./ethereumAddressResolver";
+import { IpfsStorageProvider, IpfsStorageProviderSettings } from "./storage";
+import { EthereumProofPointRegistry } from "./ethereumProofPointRegistry";
+
+/**
+ * Proof point validator
+ * Used to validate proof points.
+ */
+export class ProofPointValidator {
+  _resolver: ProofPointResolver;
+  _authenticator: ProofPointAuthenticator;
+
+  /**
+   * Construct a @ProofPointValidator for production use
+   * @param registryRootAddress The Ethereum address of a proof point registry
+   * @param ethereumProvider A provider to use for Ethereum interactions
+   * @param ipfsSettings connection settings for an IPFS node to use for storage
+   * @param registryAddress Optional, if provided will be used as the registry address, meaning
+   * the address will not be looked up in the storage contract, saving one Ethereum call.
+   * @returns a ready to use @ProofPointValidator capable of validating all types of proof point.
+   */
+  public static async production(
```

Comment:
See earlier point about this method name.

Reply:
Renamed to init